### PR TITLE
fix: failing e2e and allow running workflow on PRs

### PR DESCRIPTION
**Pull request by cmsparks**

### CODE REVIEW

File: .github/workflows/live.yaml [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+            tools/goose/go.sum
+            tools/omnaralint/go.sum
+
+      - run: make test-live-web
+        env:
+          EXA_API_KEY: ${{ secrets.EXA_API_KEY }}
+
+  live:
+    name: Live Provider Tests
+    if: github.event_name != 'pull_request' || contains(github.event.pull_request.labels.*.name, 'live-tests')
```
Comment (asherdale) [inline .github/workflows/live.yaml]: edge case - should we fail with a clear message when `live-tests` is requested on a fork PR? GitHub withholds repository secrets, so this currently skips every provider test but reports success, making it look like the requested tests passed

File: .github/workflows/live.yaml [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+  contents: read
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: ${{ github.event_name == 'pull_request' }}
+
+jobs:
+  live-web:
+    name: Live Web Tests
+    if: github.event_name != 'pull_request' || contains(github.event.pull_request.labels.*.name, 'live-tests')
```
[the remark this answers is not in the document]
Comment (asherdale) [reply .github/workflows/live.yaml]: this is worth fixing too

**Review by asherdale — APPROVED** [review]
